NewForce-Cohort-9/newforce-c9-tabloid-cli-TabloidCLI
# Dfbranch

**Description**
This PR implements CLI logic to manage posts within the TabloidCLI application. The following changes have been made:

PostTagRepository.cs: A new repository class that handles the many-to-many relationship between posts and tags. This class allows for adding, viewing, and removing tags associated with posts.

PostDetailManager.cs: A new UI manager that provides detailed interaction with posts, including viewing post details, adding tags to posts, and removing tags from posts.

Post Management Enhancements: The logic for post management has been enhanced to allow users to view post details, add tags to a post, and remove tags from a post. These features have been integrated into the Post Management menu.

**Type of Change**
New feature (non-breaking change which adds functionality)

**Testing Instructions**

1. Start the Application:
Confirm your on dfbranch
Open the TabloidCLI.sln solution in your IDE.
Run the program.

3. Access the Post Management Menu:
From the main console menu, enter 4 to access the Post Management menu.

4. Test Post Management Features:
- View Posts: Press 1 to view a list of posts.
- Add Post: Press 2 to add a new post. Follow the prompts to enter the post details.
- Remove Post: Press 3 to remove an existing post.
- Edit Post: Press 4 to edit the details of an existing post.
- View Post Details: Press 5 to view the details of a specific post.
     _Inside the Post Details menu:_
        - Press 1 to view the post's details, including any tags associated with the post.
        - Press 2 to add a tag to the post. Choose from the list of available tags.
        - Press 3 to remove a tag from the post. Select the tag you wish to remove.
- Return to Main Menu: Press 0 to return to the main menu.

**Checklist:**
